### PR TITLE
Fix partial set of MediaBrowser.Controller/Entities warnings

### PR DESCRIPTION
**Pull request by richlander**
Supports #2149

### CODE REVIEW

File: MediaBrowser.Controller/Entities/AggregateFolder.cs [hunk 1]
```diff
@@ -23,6 +23,9 @@ namespace MediaBrowser.Controller.Entities
     public class AggregateFolder : Folder
     {
         private bool _requiresRefresh;
+        private Guid[] _childrenIds = null;
+        private readonly object _childIdsLock = new object();
+
```
Comment (crobibero) [inline MediaBrowser.Controller/Entities/AggregateFolder.cs:28]:
```suggestion
```

File: MediaBrowser.Controller/Entities/AggregateFolder.cs [hunk 2]
```diff
@@ -55,15 +53,17 @@ public override bool CanDelete()
         public override string[] PhysicalLocations => PhysicalLocationsList;
 
         public string[] PhysicalLocationsList { get; set; }
+        public override bool CanDelete()
+        {
+            return false;
+        }
+
```
Comment (crobibero) [inline MediaBrowser.Controller/Entities/AggregateFolder.cs:60]:
```suggestion
```

File: MediaBrowser.Controller/Entities/Year.cs [hunk 3]
```diff
@@ -15,6 +15,17 @@ namespace MediaBrowser.Controller.Entities
...
+        [JsonIgnore]
+        public override bool SupportsAncestors => false;
+
+        public override bool CanDelete()
+        {
+            return false;
+        }
+
+        [JsonIgnore]
+        public override bool SupportsPeople => false;
```
Comment (crobibero) [inline MediaBrowser.Controller/Entities/Year.cs:27]: I believe this should be above `CanDelete`?

**Comment by richlander** [comment]
I fixed a bunch more. It's really hard to find a stopping place w/o fixing the entire project. I think we should either stop here for this PR or I'm happy to complete the whole directory.

**Review by crobibero — APPROVED** [review]

File: MediaBrowser.Controller/Entities/Folder.cs [hunk 4]
```diff
@@ -258,6 +260,7 @@ public override bool IsVisible(User user)
         /// Loads our children.  Validation will occur externally.
         /// We want this synchronous.
         /// </summary>
+        /// <returns>Returns cached children</returns>
```
Comment (cvium) [inline MediaBrowser.Controller/Entities/Folder.cs:263]: Not actually cached...
```suggestion
        /// <returns>Returns children</returns>
```

**Review by cvium — APPROVED** [review]

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**